vllm/executor/msgspec_utils.py

```diff
@@ -0,0 +1,36 @@
+from array import array
+from typing import Any, Type
+
+from vllm.sequence import SequenceData
+
+
+def encode_hook(obj: Any) -> Any:
+    """Custom msgspec enc hook that supports array types.
+
+    See https://jcristharif.com/msgspec/api.html#msgspec.msgpack.Encoder
+    """
+    if isinstance(obj, array):
+        return obj.tobytes()
+    if isinstance(obj, SequenceData):
+        # This can be reconstructed from __post_init__.
+        obj._prompt_token_ids_tuple = tuple()
+        obj._cached_all_token_ids = []
+        obj._new_appended_tokens = []
```

Comment:
no return value?

Reply:
hmm actually I found this doesn't do anything because SequenceData is already serializable... removed